api/src/main/scala/uk/ac/wellcome/platform/api/controllers/WorksController.scala

```diff
@@ -75,35 +76,45 @@ class WorksController @Inject()(@Flag("api.prefix") apiPrefix: String,
     } { request: MultipleResultsRequest =>
       val pageSize = request.pageSize.getOrElse((defaultPageSize))
 
-      val works = request.query match {
-        case Some(queryString) =>
-          worksService.searchWorks(
-            queryString,
-            pageSize = pageSize,
-            pageNumber = request.page,
-            includes = request.includes.getOrElse(WorksIncludes()),
-            index = request._index
-          )
-        case None =>
-          worksService.listWorks(
-            pageSize = pageSize,
-            pageNumber = request.page,
-            includes = request.includes.getOrElse(WorksIncludes()),
-            index = request._index
+      if (pageSize * request.page > esMaxResultWindow) {
```

Comment:
Not sure whether it's better to just allow the bad query to run and catch that exception in the error mapper.

Reply:
I think easier to catch upfront – if run the query and catch the exception, we’re paying the penalty for a query we could skip, and have to work out if the exception is this particular failure mode, or something different (which should still raise as a 500).

Reply:
Yeah, but this does move logic about what an appropriate search size is into this controller when actually ES is controlling that. Also having to work out  if the exception is this particular failure mode - because it means we can be sure this is actually the error that has occurred.

Reply:
Mmm. Yeah, okay, convinced – time to bring back the custom exception mapper!

And it occurs to me we might want to surface other stuff as 400 errors in future, e.g. malformed query syntax.